src/auth/googleauth.ts

```diff
@@ -286,6 +302,42 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     return this._findProjectIdPromise;
   }
 
+  async #getUniverseFromMetadataServer() {
+    if (!(await this._checkIsGCE())) return;
+
+    let universeDomain: string;
+
+    try {
+      universeDomain = await gcpMetadata.universe('universe_domain');
+      universeDomain ||= DEFAULT_UNIVERSE;
+    } catch (e) {
+      if (e instanceof GaxiosError && e.status === 404) {
```

Comment:
What is the situation that we would expect a 404 here? Are there any other error codes we would want to ignore and just use default universe?

Reply:
(Provided background offline)